src/ucp/core/ucp_listener.c

```diff
@@ -219,23 +268,16 @@ ucs_status_t ucp_listener_create(ucp_worker_h worker,
             continue;
         }
 
-        listener = ucs_calloc(1, sizeof(*listener), "ucp_listener");
-        if (listener == NULL) {
+        tmp = ucs_realloc(listener->wifaces,
```

Comment:
maybe create a map of mds, and then alloc wifaces just once

Reply:
@brminich @evgeny-leksikov The amount of allocated wifaces also depends on their successful iface_open and iface_init since only after they succeed, i increase sockaddr_tls which is the number of allocated wifaces.
if i create a bitmap, i may allocate more than required since some of them may not pass the iface open and init stages. 
WDYT?

Reply:
I think that's good approach, but why them may not pass initialization if you take into account capabilities on creation of the bitmap?

Reply:
@evgeny-leksikov these functions ucp_worker_iface_open(), ucp_worker_iface_init() may return a status that is not UCX_OK becuase of internal functions in them, failing memory allocations etc. if their return value is ucs_status_t then i think we need to handle it. we cant assume it would always return UCS_OK...

Reply:
IMO in this case `ucp_listener_create` should fail as well

Reply:
if we now support several listening transports, why not try the next one if the previous one failed?

Reply:
it is not really a major comment. IMO, you can leave it as it is, or do calloc(all_tls) and then shrink by real sock_tls count using realloc

Reply:
> if we now support several listening transports, why not try the next one if the previous one failed?

because one client should be able to connect by rdmacm but other one by sock_cm. for example if they are from in different subnets. I don't think that we need to hide lack of memory (for example) by disabling some components.

Reply:
@evgeny-leksikov why fail both clients then if only one won't be able to connect?

Reply:
OK, should we somehow to report user that some transports initialization failed?
IMO warning is not enough because it should be handled programmatically.